### PR TITLE
Adds reproducible build system & build commands

### PR DESCRIPTION
**Pull request by jiyuu-jin**

### CODE REVIEW

File: internal/build/reproducible.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+	if dockerConfig.BuildCommand != "" {
+		dockerfile += fmt.Sprintf("\nRUN %s", dockerConfig.BuildCommand)
+	}
+
+	if dockerConfig.InstallCommand != "" {
+		dockerfile += fmt.Sprintf("\nRUN %s", dockerConfig.InstallCommand)
+	}
+
+	os.WriteFile(dockerConfig.Path, []byte(dockerfile), 0644)
```
Comment (nasdf) [inline internal/build/reproducible.go:32]: Return the error from here.

File: internal/build/reproducible.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+	}
+	hostPath := filepath.Join(cwd, out)
+	containerPath := fmt.Sprintf("valist-build:/opt/build/%s", out)
+
+	// fmt.Println(hostPath)
+	// fmt.Println(containerPath)
+
+	// If output is a single directory, remove the directory
+	if _, err := os.Stat(hostPath); !os.IsNotExist(err) {
+		err := os.Remove(hostPath)
```
Comment (nasdf) [inline internal/build/reproducible.go:59]: Might want `os.RemoveAll` here

File: internal/build/reproducible.go [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+	err = createCmd.Run()
+	if err != nil {
+		return err
+	}
+
+	cpCmd := exec.Command("docker", "cp", containerPath, hostPath)
+	cpCmd.Stdout = os.Stdout
+	cpCmd.Stderr = os.Stderr
+
+	cpCmd.Run()
```
Comment (nasdf) [inline internal/build/reproducible.go:77]: Need error check here

File: internal/build/reproducible.go [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+	if err != nil {
+		return err
+	}
+
+	rmCmd := exec.Command("docker", "rm", "valist-build")
+	rmCmd.Stdout = os.Stdout
+	rmCmd.Stderr = os.Stderr
+	rmCmd.Run()
```
Comment (nasdf) [inline internal/build/reproducible.go:85]: Need error check here.  For the final error check just return the error value.

```go
return rmCmd.Run()
```

File: internal/build/reproducible.go [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+		if err != nil {
+			return err
+		}
+	}
+
+	createCmd := exec.Command("docker", "create", "--name=valist-build", "valist-build")
+	createCmd.Stdout = os.Stdout
+	createCmd.Stderr = os.Stderr
+	err = createCmd.Run()
+	if err != nil {
```
Comment (nasdf) [inline internal/build/reproducible.go:69]: This is a good spot to inline the error check.

```go
if err := createCmd.Run(); err != nil {
    return err
}
```

File: internal/build/reproducible.go [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+	os.WriteFile(dockerConfig.Path, []byte(dockerfile), 0644)
+}
+
+func Create(imageTag string) error {
+	cmd := exec.Command("docker", "build", ".", "-t", imageTag, "--progress=plain")
+	cmd.Env = os.Environ()
+	cmd.Env = append(cmd.Env, "DOCKER_BUILDKIT=1")
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	cmd.Run()
```
Comment (nasdf) [inline internal/build/reproducible.go:41]: Need to return error from here.

**Review by nasdf — COMMENTED** [review]
Looking good!  Just need to make sure all errors are checked and returned.

File: internal/build/reproducible.go (hunk 5 above)
Comment (jiyuu-jin) [reply internal/build/reproducible.go:69]: I love this syntax , so clean!

File: internal/build/reproducible.go (hunk 6 above)
Comment (jiyuu-jin) [reply internal/build/reproducible.go:41]: Got it!

File: internal/build/reproducible.go (hunk 3 above)
Comment (jiyuu-jin) [reply internal/build/reproducible.go:77]: Good catch, added it!

File: internal/build/reproducible.go (hunk 2 above)
Comment (jiyuu-jin) [reply internal/build/reproducible.go:59]: Great tip! Hopefully this will prevent any weird things from happening with the folder cleanup.

**Review by nasdf — COMMENTED** [review]
A couple more small changes.

**Review by nasdf — APPROVED** [review]
Looks great!

**Pull request merged**